lib/kadm5/get_s.c

```diff
@@ -323,6 +324,12 @@ kadm5_s_get_principal(void *server_handle,
 	    _krb5_put_int(buf, last_pw_expire, sizeof(buf));
 	    ret = add_tl_data(out, KRB5_TL_LAST_PWD_CHANGE, buf, sizeof(buf));
 	}
+        if (ret == 0)
+            ret = hdb_entry_get_krb5_config(&ent.entry, &krb5_config);
+        if (ret == 0 && krb5_config.length) {
+            ret = add_tl_data(out, KRB5_TL_KRB5_CONFIG, krb5_config.data,
```

Comment:
XXX Make `add_tl_data()` check for `tl_data_length` overflow.

Reply:
On the wire we use 32-bit signed ints for TLs lengths.  We should just change the struct and the protocol to use unsigned 32-bit lengths and just break the ABI.  (And also we should limit the size of a TL to 2^18 or some such.